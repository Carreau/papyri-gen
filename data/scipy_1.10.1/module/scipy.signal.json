{
  "aliases": [
    "scipy.signal"
  ],
  "arbitrary": [
    {
      "children": [],
      "level": 0,
      "target": null,
      "title": "Signal processing (:mod:`scipy.signal`)",
      "type": "Section"
    },
    {
      "children": [
        {
          "type": "code",
          "value": ".. autosummary:: \n    :toctree:generated/\n    convolve           -- N-D convolution.\n    correlate          -- N-D correlation.\n    fftconvolve        -- N-D convolution using the FFT.\n    oaconvolve         -- N-D convolution using the overlap-add method.\n    convolve2d         -- 2-D convolution (more options).\n    correlate2d        -- 2-D correlation (more options).\n    sepfir2d           -- Convolve with a 2-D separable FIR filter.\n    choose_conv_method -- Chooses faster of FFT and direct convolution methods.\n    correlation_lags   -- Determines lag indices for 1D cross-correlation."
        }
      ],
      "level": 1,
      "target": null,
      "title": "Convolution",
      "type": "Section"
    },
    {
      "children": [
        {
          "type": "code",
          "value": ".. autosummary:: \n    :toctree:generated/\n    bspline        -- B-spline basis function of order n.\n    cubic          -- B-spline basis function of order 3.\n    quadratic      -- B-spline basis function of order 2.\n    gauss_spline   -- Gaussian approximation to the B-spline basis function.\n    cspline1d      -- Coefficients for 1-D cubic (3rd order) B-spline.\n    qspline1d      -- Coefficients for 1-D quadratic (2nd order) B-spline.\n    cspline2d      -- Coefficients for 2-D cubic (3rd order) B-spline.\n    qspline2d      -- Coefficients for 2-D quadratic (2nd order) B-spline.\n    cspline1d_eval -- Evaluate a cubic spline at the given points.\n    qspline1d_eval -- Evaluate a quadratic spline at the given points.\n    spline_filter  -- Smoothing spline (cubic) filtering of a rank-2 array."
        }
      ],
      "level": 1,
      "target": null,
      "title": "B-splines",
      "type": "Section"
    },
    {
      "children": [
        {
          "type": "code",
          "value": ".. autosummary:: \n    :toctree:generated/\n    order_filter  -- N-D order filter.\n    medfilt       -- N-D median filter.\n    medfilt2d     -- 2-D median filter (faster).\n    wiener        -- N-D Wiener filter.\n\n    symiirorder1  -- 2nd-order IIR filter (cascade of first-order systems).\n    symiirorder2  -- 4th-order IIR filter (cascade of second-order systems).\n    lfilter       -- 1-D FIR and IIR digital linear filtering.\n    lfiltic       -- Construct initial conditions for `lfilter`.\n    lfilter_zi    -- Compute an initial state zi for the lfilter function that\n                  -- corresponds to the steady state of the step response.\n    filtfilt      -- A forward-backward filter.\n    savgol_filter -- Filter a signal using the Savitzky-Golay filter.\n\n    deconvolve    -- 1-D deconvolution using lfilter.\n\n    sosfilt       -- 1-D IIR digital linear filtering using\n                  -- a second-order sections filter representation.\n    sosfilt_zi    -- Compute an initial state zi for the sosfilt function that\n                  -- corresponds to the steady state of the step response.\n    sosfiltfilt   -- A forward-backward filter for second-order sections.\n    hilbert       -- Compute 1-D analytic signal, using the Hilbert transform.\n    hilbert2      -- Compute 2-D analytic signal, using the Hilbert transform.\n\n    decimate      -- Downsample a signal.\n    detrend       -- Remove linear and/or constant trends from data.\n    resample      -- Resample using Fourier method.\n    resample_poly -- Resample using polyphase filtering method.\n    upfirdn       -- Upsample, apply FIR filter, downsample."
        }
      ],
      "level": 1,
      "target": null,
      "title": "Filtering",
      "type": "Section"
    },
    {
      "children": [
        {
          "type": "code",
          "value": ".. autosummary:: \n    :toctree:generated/\n    bilinear      -- Digital filter from an analog filter using\n                     -- the bilinear transform.\n    bilinear_zpk  -- Digital filter from an analog filter using\n                     -- the bilinear transform.\n    findfreqs     -- Find array of frequencies for computing filter response.\n    firls         -- FIR filter design using least-squares error minimization.\n    firwin        -- Windowed FIR filter design, with frequency response\n                     -- defined as pass and stop bands.\n    firwin2       -- Windowed FIR filter design, with arbitrary frequency\n                     -- response.\n    freqs         -- Analog filter frequency response from TF coefficients.\n    freqs_zpk     -- Analog filter frequency response from ZPK coefficients.\n    freqz         -- Digital filter frequency response from TF coefficients.\n    freqz_zpk     -- Digital filter frequency response from ZPK coefficients.\n    sosfreqz      -- Digital filter frequency response for SOS format filter.\n    gammatone     -- FIR and IIR gammatone filter design.\n    group_delay   -- Digital filter group delay.\n    iirdesign     -- IIR filter design given bands and gains.\n    iirfilter     -- IIR filter design given order and critical frequencies.\n    kaiser_atten  -- Compute the attenuation of a Kaiser FIR filter, given\n                     -- the number of taps and the transition width at\n                     -- discontinuities in the frequency response.\n    kaiser_beta   -- Compute the Kaiser parameter beta, given the desired\n                     -- FIR filter attenuation.\n    kaiserord     -- Design a Kaiser window to limit ripple and width of\n                     -- transition region.\n    minimum_phase -- Convert a linear phase FIR filter to minimum phase.\n    savgol_coeffs -- Compute the FIR filter coefficients for a Savitzky-Golay\n                     -- filter.\n    remez         -- Optimal FIR filter design.\n\n    unique_roots  -- Unique roots and their multiplicities.\n    residue       -- Partial fraction expansion of b(s) / a(s).\n    residuez      -- Partial fraction expansion of b(z) / a(z).\n    invres        -- Inverse partial fraction expansion for analog filter.\n    invresz       -- Inverse partial fraction expansion for digital filter.\n    BadCoefficients  -- Warning on badly conditioned filter coefficients."
        },
        {
          "children": [
            {
              "type": "text",
              "value": "Lower-level filter design functions:"
            }
          ],
          "type": "paragraph"
        },
        {
          "type": "code",
          "value": ".. autosummary:: \n    :toctree:generated/\n    abcd_normalize -- Check state-space matrices and ensure they are rank-2.\n    band_stop_obj  -- Band Stop Objective Function for order minimization.\n    besselap       -- Return (z,p,k) for analog prototype of Bessel filter.\n    buttap         -- Return (z,p,k) for analog prototype of Butterworth filter.\n    cheb1ap        -- Return (z,p,k) for type I Chebyshev filter.\n    cheb2ap        -- Return (z,p,k) for type II Chebyshev filter.\n    cmplx_sort     -- Sort roots based on magnitude.\n    ellipap        -- Return (z,p,k) for analog prototype of elliptic filter.\n    lp2bp          -- Transform a lowpass filter prototype to a bandpass filter.\n    lp2bp_zpk      -- Transform a lowpass filter prototype to a bandpass filter.\n    lp2bs          -- Transform a lowpass filter prototype to a bandstop filter.\n    lp2bs_zpk      -- Transform a lowpass filter prototype to a bandstop filter.\n    lp2hp          -- Transform a lowpass filter prototype to a highpass filter.\n    lp2hp_zpk      -- Transform a lowpass filter prototype to a highpass filter.\n    lp2lp          -- Transform a lowpass filter prototype to a lowpass filter.\n    lp2lp_zpk      -- Transform a lowpass filter prototype to a lowpass filter.\n    normalize      -- Normalize polynomial representation of a transfer function."
        }
      ],
      "level": 1,
      "target": null,
      "title": "Filter design",
      "type": "Section"
    },
    {
      "children": [
        {
          "type": "code",
          "value": ".. autosummary:: \n    :toctree:generated/\n    butter -- Butterworth\n    buttord\n    cheby1 -- Chebyshev Type I\n    cheb1ord\n    cheby2 -- Chebyshev Type II\n    cheb2ord\n    ellip -- Elliptic (Cauer)\n    ellipord\n    bessel -- Bessel (no order selection available -- try butterod)\n    iirnotch      -- Design second-order IIR notch digital filter.\n    iirpeak       -- Design second-order IIR peak (resonant) digital filter.\n    iircomb       -- Design IIR comb filter."
        }
      ],
      "level": 1,
      "target": null,
      "title": "Matlab-style IIR filter design",
      "type": "Section"
    },
    {
      "children": [
        {
          "type": "code",
          "value": ".. autosummary:: \n    :toctree:generated/\n    lti              -- Continuous-time linear time invariant system base class.\n    StateSpace       -- Linear time invariant system in state space form.\n    TransferFunction -- Linear time invariant system in transfer function form.\n    ZerosPolesGain   -- Linear time invariant system in zeros, poles, gain form.\n    lsim             -- Continuous-time simulation of output to linear system.\n    lsim2            -- Like lsim, but `scipy.integrate.odeint` is used.\n    impulse          -- Impulse response of linear, time-invariant (LTI) system.\n    impulse2         -- Like impulse, but `scipy.integrate.odeint` is used.\n    step             -- Step response of continuous-time LTI system.\n    step2            -- Like step, but `scipy.integrate.odeint` is used.\n    freqresp         -- Frequency response of a continuous-time LTI system.\n    bode             -- Bode magnitude and phase data (continuous-time LTI)."
        }
      ],
      "level": 1,
      "target": null,
      "title": "Continuous-time linear systems",
      "type": "Section"
    },
    {
      "children": [
        {
          "type": "code",
          "value": ".. autosummary:: \n    :toctree:generated/\n    dlti             -- Discrete-time linear time invariant system base class.\n    StateSpace       -- Linear time invariant system in state space form.\n    TransferFunction -- Linear time invariant system in transfer function form.\n    ZerosPolesGain   -- Linear time invariant system in zeros, poles, gain form.\n    dlsim            -- Simulation of output to a discrete-time linear system.\n    dimpulse         -- Impulse response of a discrete-time LTI system.\n    dstep            -- Step response of a discrete-time LTI system.\n    dfreqresp        -- Frequency response of a discrete-time LTI system.\n    dbode            -- Bode magnitude and phase data (discrete-time LTI)."
        }
      ],
      "level": 1,
      "target": null,
      "title": "Discrete-time linear systems",
      "type": "Section"
    },
    {
      "children": [
        {
          "type": "code",
          "value": ".. autosummary:: \n    :toctree:generated/\n    tf2zpk        -- Transfer function to zero-pole-gain.\n    tf2sos        -- Transfer function to second-order sections.\n    tf2ss         -- Transfer function to state-space.\n    zpk2tf        -- Zero-pole-gain to transfer function.\n    zpk2sos       -- Zero-pole-gain to second-order sections.\n    zpk2ss        -- Zero-pole-gain to state-space.\n    ss2tf         -- State-pace to transfer function.\n    ss2zpk        -- State-space to pole-zero-gain.\n    sos2zpk       -- Second-order sections to zero-pole-gain.\n    sos2tf        -- Second-order sections to transfer function.\n    cont2discrete -- Continuous-time to discrete-time LTI conversion.\n    place_poles   -- Pole placement."
        }
      ],
      "level": 1,
      "target": null,
      "title": "LTI representations",
      "type": "Section"
    },
    {
      "children": [
        {
          "type": "code",
          "value": ".. autosummary:: \n    :toctree:generated/\n    chirp        -- Frequency swept cosine signal, with several freq functions.\n    gausspulse   -- Gaussian modulated sinusoid.\n    max_len_seq  -- Maximum length sequence.\n    sawtooth     -- Periodic sawtooth.\n    square       -- Square wave.\n    sweep_poly   -- Frequency swept cosine signal; freq is arbitrary polynomial.\n    unit_impulse -- Discrete unit impulse."
        }
      ],
      "level": 1,
      "target": null,
      "title": "Waveforms",
      "type": "Section"
    },
    {
      "children": [
        {
          "children": [
            {
              "type": "text",
              "value": "For window functions, see the "
            },
            {
              "anchor": null,
              "exists": true,
              "kind": "module",
              "reference": {
                "kind": "module",
                "module": "scipy",
                "path": "scipy.signal.windows",
                "type": "RefInfo",
                "version": "1.10.1"
              },
              "type": "Link",
              "value": "scipy.signal.windows"
            },
            {
              "type": "text",
              "value": " namespace."
            }
          ],
          "type": "paragraph"
        },
        {
          "children": [
            {
              "type": "text",
              "value": "In the "
            },
            {
              "anchor": null,
              "exists": true,
              "kind": "module",
              "reference": {
                "kind": "module",
                "module": "scipy",
                "path": "scipy.signal",
                "type": "RefInfo",
                "version": "1.10.1"
              },
              "type": "Link",
              "value": "scipy.signal"
            },
            {
              "type": "text",
              "value": " namespace, there is a convenience function to obtain these windows by name:"
            }
          ],
          "type": "paragraph"
        },
        {
          "type": "code",
          "value": ".. autosummary:: \n    :toctree:generated/\n    get_window -- Return a window of a given length and type."
        }
      ],
      "level": 1,
      "target": null,
      "title": "Window functions",
      "type": "Section"
    },
    {
      "children": [
        {
          "type": "code",
          "value": ".. autosummary:: \n    :toctree:generated/\n    cascade      -- Compute scaling function and wavelet from coefficients.\n    daub         -- Return low-pass.\n    morlet       -- Complex Morlet wavelet.\n    qmf          -- Return quadrature mirror filter from low-pass.\n    ricker       -- Return ricker wavelet.\n    morlet2      -- Return Morlet wavelet, compatible with cwt.\n    cwt          -- Perform continuous wavelet transform."
        }
      ],
      "level": 1,
      "target": null,
      "title": "Wavelets",
      "type": "Section"
    },
    {
      "children": [
        {
          "type": "code",
          "value": ".. autosummary:: \n    :toctree:generated/\n    argrelmin        -- Calculate the relative minima of data.\n    argrelmax        -- Calculate the relative maxima of data.\n    argrelextrema    -- Calculate the relative extrema of data.\n    find_peaks       -- Find a subset of peaks inside a signal.\n    find_peaks_cwt   -- Find peaks in a 1-D array with wavelet transformation.\n    peak_prominences -- Calculate the prominence of each peak in a signal.\n    peak_widths      -- Calculate the width of each peak in a signal."
        }
      ],
      "level": 1,
      "target": null,
      "title": "Peak finding",
      "type": "Section"
    },
    {
      "children": [
        {
          "type": "code",
          "value": ".. autosummary:: \n    :toctree:generated/\n    periodogram    -- Compute a (modified) periodogram.\n    welch          -- Compute a periodogram using Welch's method.\n    csd            -- Compute the cross spectral density, using Welch's method.\n    coherence      -- Compute the magnitude squared coherence, using Welch's method.\n    spectrogram    -- Compute the spectrogram.\n    lombscargle    -- Computes the Lomb-Scargle periodogram.\n    vectorstrength -- Computes the vector strength.\n    stft           -- Compute the Short Time Fourier Transform.\n    istft          -- Compute the Inverse Short Time Fourier Transform.\n    check_COLA     -- Check the COLA constraint for iSTFT reconstruction.\n    check_NOLA     -- Check the NOLA constraint for iSTFT reconstruction."
        }
      ],
      "level": 1,
      "target": null,
      "title": "Spectral analysis",
      "type": "Section"
    },
    {
      "children": [
        {
          "type": "code",
          "value": ".. autosummary:: \n    :toctree:generated/\n    czt - Chirp z-transform convenience function\n    zoom_fft - Zoom FFT convenience function\n    CZT - Chirp z-transform function generator\n    ZoomFFT - Zoom FFT function generator\n    czt_points - Output the z-plane points sampled by a chirp z-transform"
        },
        {
          "children": [
            {
              "type": "text",
              "value": "The functions are simpler to use than the classes, but are less efficient when using the same transform on many arrays of the same length, since they repeatedly generate the same chirp signal with every call.  In these cases, use the classes to create a reusable function instead."
            }
          ],
          "type": "paragraph"
        }
      ],
      "level": 1,
      "target": null,
      "title": "Chirp Z-transform and Zoom FFT",
      "type": "Section"
    }
  ],
  "content": {
    "Attributes": {
      "children": [],
      "level": 0,
      "target": null,
      "title": null,
      "type": "Section"
    },
    "Extended Summary": {
      "children": [],
      "level": 0,
      "target": null,
      "title": null,
      "type": "Section"
    },
    "Methods": {
      "children": [],
      "level": 0,
      "target": null,
      "title": null,
      "type": "Section"
    },
    "Notes": {
      "children": [],
      "level": 0,
      "target": null,
      "title": null,
      "type": "Section"
    },
    "Other Parameters": {
      "children": [],
      "level": 0,
      "target": null,
      "title": null,
      "type": "Section"
    },
    "Parameters": {
      "children": [],
      "level": 0,
      "target": null,
      "title": null,
      "type": "Section"
    },
    "Raises": {
      "children": [],
      "level": 0,
      "target": null,
      "title": null,
      "type": "Section"
    },
    "Receives": {
      "children": [],
      "level": 0,
      "target": null,
      "title": null,
      "type": "Section"
    },
    "Returns": {
      "children": [],
      "level": 0,
      "target": null,
      "title": null,
      "type": "Section"
    },
    "Summary": {
      "children": [
        {
          "children": [
            {
              "type": "text",
              "value": "To remove in the future \u2013\u2013 scipy.signal"
            }
          ],
          "type": "paragraph"
        }
      ],
      "level": 0,
      "target": null,
      "title": null,
      "type": "Section"
    },
    "Warnings": {
      "children": [],
      "level": 0,
      "target": null,
      "title": null,
      "type": "Section"
    },
    "Warns": {
      "children": [],
      "level": 0,
      "target": null,
      "title": null,
      "type": "Section"
    },
    "Yields": {
      "children": [],
      "level": 0,
      "target": null,
      "title": null,
      "type": "Section"
    }
  },
  "example_section_data": {
    "children": [],
    "level": 0,
    "target": null,
    "title": null,
    "type": "Section"
  },
  "item_file": "/scipy/signal/__init__.py",
  "item_line": 0,
  "item_type": "<class 'module'>",
  "ordered_sections": [
    "Summary",
    "Extended Summary"
  ],
  "references": null,
  "see_also": [],
  "signature": {
    "type": "Signature",
    "value": null
  },
  "type": "DocBlob"
}