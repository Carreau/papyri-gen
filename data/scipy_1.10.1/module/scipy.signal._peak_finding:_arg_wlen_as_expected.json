{
  "aliases": [
    "scipy.signal._peak_finding._arg_wlen_as_expected"
  ],
  "arbitrary": [],
  "content": {
    "Attributes": {
      "children": [],
      "level": 0,
      "target": null,
      "title": null,
      "type": "Section"
    },
    "Extended Summary": {
      "children": [
        {
          "children": [
            {
              "type": "text",
              "value": "Used in "
            },
            {
              "domain": null,
              "role": null,
              "type": "Directive",
              "value": "peak_prominences"
            },
            {
              "type": "text",
              "value": " and "
            },
            {
              "domain": null,
              "role": null,
              "type": "Directive",
              "value": "peak_widths"
            },
            {
              "type": "text",
              "value": "."
            }
          ],
          "type": "paragraph"
        }
      ],
      "level": 0,
      "target": null,
      "title": null,
      "type": "Section"
    },
    "Methods": {
      "children": [],
      "level": 0,
      "target": null,
      "title": null,
      "type": "Section"
    },
    "Notes": {
      "children": [],
      "level": 0,
      "target": null,
      "title": null,
      "type": "Section"
    },
    "Other Parameters": {
      "children": [],
      "level": 0,
      "target": null,
      "title": null,
      "type": "Section"
    },
    "Parameters": {
      "children": [],
      "level": 0,
      "target": null,
      "title": null,
      "type": "Section"
    },
    "Raises": {
      "children": [],
      "level": 0,
      "target": null,
      "title": null,
      "type": "Section"
    },
    "Receives": {
      "children": [],
      "level": 0,
      "target": null,
      "title": null,
      "type": "Section"
    },
    "Returns": {
      "children": [
        {
          "children": [
            {
              "desc": [
                {
                  "children": [
                    {
                      "type": "text",
                      "value": "The original "
                    },
                    {
                      "anchor": null,
                      "exists": true,
                      "kind": "local",
                      "reference": {
                        "kind": "local",
                        "module": null,
                        "path": "value",
                        "type": "RefInfo",
                        "version": null
                      },
                      "type": "Link",
                      "value": "value"
                    },
                    {
                      "type": "text",
                      "value": " rounded up to an integer or -1 if "
                    },
                    {
                      "anchor": null,
                      "exists": true,
                      "kind": "local",
                      "reference": {
                        "kind": "local",
                        "module": null,
                        "path": "value",
                        "type": "RefInfo",
                        "version": null
                      },
                      "type": "Link",
                      "value": "value"
                    },
                    {
                      "type": "text",
                      "value": " was None."
                    }
                  ],
                  "type": "paragraph"
                }
              ],
              "param": "value",
              "type": "Param",
              "type_": "np.intp"
            }
          ],
          "type": "Parameters"
        }
      ],
      "level": 0,
      "target": null,
      "title": null,
      "type": "Section"
    },
    "Summary": {
      "children": [
        {
          "children": [
            {
              "type": "text",
              "value": "Ensure argument "
            },
            {
              "anchor": null,
              "exists": true,
              "kind": "module",
              "reference": {
                "kind": "module",
                "module": "scipy",
                "path": "scipy.signal._peak_finding:_arg_wlen_as_expected",
                "type": "RefInfo",
                "version": "1.10.1"
              },
              "type": "Link",
              "value": "wlen"
            },
            {
              "type": "text",
              "value": " is of type "
            },
            {
              "domain": null,
              "role": null,
              "type": "Directive",
              "value": "np.intp"
            },
            {
              "type": "text",
              "value": " and larger than 1."
            }
          ],
          "type": "paragraph"
        }
      ],
      "level": 0,
      "target": null,
      "title": null,
      "type": "Section"
    },
    "Warnings": {
      "children": [],
      "level": 0,
      "target": null,
      "title": null,
      "type": "Section"
    },
    "Warns": {
      "children": [],
      "level": 0,
      "target": null,
      "title": null,
      "type": "Section"
    },
    "Yields": {
      "children": [],
      "level": 0,
      "target": null,
      "title": null,
      "type": "Section"
    }
  },
  "example_section_data": {
    "children": [],
    "level": 0,
    "target": null,
    "title": null,
    "type": "Section"
  },
  "item_file": "/scipy/signal/_peak_finding.py",
  "item_line": 297,
  "item_type": "<class 'function'>",
  "ordered_sections": [
    "Summary",
    "Extended Summary",
    "Returns"
  ],
  "references": null,
  "see_also": [],
  "signature": {
    "type": "Signature",
    "value": "(value)"
  },
  "type": "DocBlob"
}